framework/src/meshgenerators/SideSetsFromPointsGenerator.C

```diff
@@ -72,28 +75,59 @@ SideSetsFromPointsGenerator::generate()
 
   for (unsigned int i = 0; i < boundary_ids.size(); ++i)
   {
-    const Elem * elem = (*pl)(_points[i]);
-
-    for (unsigned int side = 0; side < elem->n_sides(); ++side)
-    {
-      if (elem->neighbor_ptr(side))
-        continue;
+    std::set<const Elem *> candidate_elements;
+    (*pl)(_points[i], candidate_elements);
 
-      // See if this point is on this side
-      std::unique_ptr<const Elem> elem_side = elem->side_ptr(side);
+    const Elem * elem_to_flood = nullptr;
+    Point normal_to_flood;
 
-      if (elem_side->contains_point(_points[i]))
+    for (const Elem * elem : candidate_elements)
+      for (unsigned int side = 0; side < elem->n_sides(); ++side)
       {
-        // This is the side that we want to paint our sideset with
-        // First get the normal
-        const std::vector<Point> & normals = _fe_face->get_normals();
-        _fe_face->reinit(elem, side);
-
-        flood(elem, normals[0], boundary_ids[i], *mesh);
+        if (elem->neighbor_ptr(side))
+          continue;
+
+        // See if this point is on this side
+        std::unique_ptr<const Elem> elem_side = elem->side_ptr(side);
+
+        if (elem_side->contains_point(_points[i]))
+        {
+          // This is a good side to paint our sideset with.
+          // Get the normal.
+          const std::vector<Point> & normals = _fe_face->get_normals();
+          _fe_face->reinit(elem, side);
+
+          // If we *already* found a good but different side to paint
+          // our sideset with, we've got an ambiguity here.
+          if (elem_to_flood && (std::abs(1.0 - normal_to_flood * normals[0]) > _variance ||
```

Comment:
Oh, sorry, I misunderstood.  Yeah, I'm being lazy here, but in my defense it's the same laziness `flood()` already uses; I'm not adding a new failure case.

Boy, now I feel more sympathy for all the people using `ReplicatedMesh` instead of `MeshSerializer`...